core/src/main/java/jenkins/model/logging/LoggingDefinitionLauncherWrapper.java

```diff
@@ -23,10 +23,160 @@
+import org.apache.commons.io.input.NullInputStream;
+
 /**
  *
  * @author Oleg Nenashev
  */
 public class LoggingDefinitionLauncherWrapper {
-    
+
+    /**
+     * Default local launcher, doesn't do anything.
+     */
+    public static class DefaultLocalLauncher extends Launcher.DecoratedLauncher {
+        public DefaultLocalLauncher(Launcher inner) {
+            super(inner);
+        }
+    }
+
+    /**
+     * Default remote launcher which redirects all the output and error to the stream {@link LoggingMethod} provides.
+     */
+    public static class DefaultRemoteLauncher extends Launcher.DecoratedLauncher {
+        private static final NullInputStream NULL_INPUT_STREAM = new NullInputStream(0);
+        private final Run run;
+        private final LoggingMethod loggingMethod;
+
+        public DefaultRemoteLauncher(Launcher inner, Run run, LoggingMethod loggingMethod) {
+            super(inner);
+            this.run = run;
+            this.loggingMethod = loggingMethod;
+        }
+
+        @Override
+        public Proc launch(Launcher.ProcStarter ps) throws IOException {
+            final LoggingMethod.OutputStreamWrapper streamOut = loggingMethod.provideOutStream(run);
+            final LoggingMethod.OutputStreamWrapper streamErr = loggingMethod.provideErrStream(run);
+
+            // RemoteLogstashReporterStream(new CloseProofOutputStream(ps.stdout()
+            final OutputStream out = ps.stdout() == null ? null : (streamOut == null ? ps.stdout() : streamOut);
+            final OutputStream err = ps.stdout() == null ? null : (streamErr == null ? ps.stdout() : streamErr);
+            final InputStream in = (ps.stdin() == null || ps.stdin() == NULL_INPUT_STREAM) ? null : new RemoteInputStream(ps.stdin(), false);
+            final String workDir = ps.pwd() == null ? null : ps.pwd().getRemote();
+
+            // TODO: we do not reverse streams => the parameters
+            try {
+                final RemoteLaunchCallable callable = new RemoteLaunchCallable(
+                    ps.cmds(), ps.masks(), ps.envs(), in,
+                    false, out, false, err,
+                    false, ps.quiet(), workDir, listener);
+
+                return new Launcher.RemoteLauncher.ProcImpl(getChannel().call(callable));
+            } catch (InterruptedException e) {
+                throw (IOException) new InterruptedIOException().initCause(e);
+            }
+        }
+
+        private static class RemoteLaunchCallable extends
+            MasterToSlaveCallable<RemoteProcess, IOException> {
+
+            private final List<String> cmd;
+            private final boolean[] masks;
+            private final String[] env;
+            private final InputStream in;
+            private final OutputStream out;
+            private final OutputStream err;
+            private final String workDir;
+            private final TaskListener listener;
+            private final boolean reverseStdin, reverseStdout, reverseStderr;
```

Comment:
Effectively we do not use these flags (See TODO above).
Maybe we want to just drop them